Fix download page images

silverailscolo opened the pull request:
CATS and Java from local png

rhwood commented:
We should probably use (either copy or by link) one of the buttons at http://www.oracle.com/us/technologies/java/gimmejava/download-buttons/index.html#java for the download. It also appears that the Java icon we have is not one we can use per http://www.oracle.com/technetwork/java/logos-136640.html#licensees , http://www.oracle.com/us/technologies/java/java-licensing-logo-guidelines-1908204.pdf and http://www.java-forums.org/forum-lobby/40921-legal-use-java-logo.html

silverailscolo commented:
I agree, hadn't looked at those pages recently. Changed to the Download Now button and removed the unofficial img files. Did we ever apply for the Java Powered button?
Might have to document this on our [copyrights](http://jmri.sourceforge.net/Copyright.html) page.

rhwood commented:
"Oracle and Java are registered trademarks of Oracle. Sun, Sun Microsystems, the Java Coffee Cup logo, and the Duke logo are trademarks of Oracle." on the copyrights page probably covers our use of the term Java.

https://github.com/JMRI/website/search?utf8=✓&q=ico-java shows where the ico-java images are in use.

rhwood commented on download/index.shtml:
```diff
@@ -160,18 +160,16 @@
-<img width="88" height="31" border="0" style="margin: 8px;" alt="GetJava Download Button" title="GetJava" src="http://java.com/en/img/everywhere/getjava_sm.gif?cid=jdp74834">
-</a>
-    <dt><a href="http://java.com/java/download/index.jsp?cid=jdp74834">Java</a></dt>
+        <a href="http://www.java.com" target="_blank"><img src="http://download.oracle.com/technetwork/java/get-java/getjavasoftware-88x31.png" alt="Get Java Software" border="0" width="88" height="31" /></a>
```
Can you make all the download links point to http://java.com/java/download/index.jsp?cid=jdp74834 (this is the only one that does not)?

silverailscolo replied:
In fact this was the url that Oracle used behind its button. Changed it, as that requires the user to look for the actual download button/page after being sent to the Oracle web site.

bobjacobsen commented:
We have a license agreement from 2005 and 2008 authorizing the use of the Java (e.g. without "Powered" or "Compatible") logo in perpetuity, along with all the other java-related icon & image files JMRI was using at that point.  

That _doesn't_ cover any later logos, which might have conditions of their own. Has anybody gone through the licensing for new logos to confirm that we're complying?

rhwood commented:
I've been going through the Java logos, and Oracle/Sun is very specific that the plain Java logo is not to be used by any other party without specific prior permission. The Java Download button is freely reusable, but only for that purpose. I haven't looked at other logos from other corporations. Do we have our license agreements archived somewhere public or semi-public?

bobjacobsen commented:
> I've been going through the Java logos, and Oracle/Sun is very specific that the plain Java logo is not to be used by any other party without specific prior permission. 

Right.  But the [situation was different 12 years ago](http://jmri.org/k/History.shtml), hence the license agreements.  

> The Java Download button is freely reusable, but only for that purpose. 

OK. Does this update make our web pages comply with the current license? (I haven't looked) Sometimes they specific the form of the HTML code, etc.

silverailscolo commented:
From the url that was displayed in the box next to the button I picked from their [page](http://www.oracle.com/us/technologies/java/gimmejava/download-buttons/index.html#java) I removed 2 cryptic ``<img foo>`` tags that are absent in the 3 alternatives. Furthermore we link not to the Oracle Java front page but the Java download page, as motivated above.

bobjacobsen commented:
We need to be cafe full with the license terms on http://www.oracle.com/us/technologies/java/gimmejava/download-buttons/index.html#java  I read that as requiring (2nd bullet) usage in a particular way. Will do some checking, which might take a couple days.

bobjacobsen commented:
I think we have two choices:

- Leave the icons and links as they were - we clearly have the right authorizations to do that

- Use the new download icons (documented here http://www.oracle.com/us/technologies/java/gimmejava/download-buttons/index.html#java) with a link to http://www.java.com Note that this includes bringing the icon from http://download.oracle.com/technetwork/java/get-java/ in the HTML, _not_ bringing it into JMRI Git.

We do not have agreements in place that would allow us to use the “Java Powered” or “Java Compatible” logos.

silverailscolo commented:
Thanks. As of today http://www.java.com provides easy access to start the download, so keeping within the advertised rule for that URL seems OK. Displaying a current Java image (colors) on our help page is the improvement in this PR. The copy of the button.png in github should be removed to comply.
@bobjacobsen Would we lose the authorization for the use of the old (Java) icon if we don't show it?

bobjacobsen commented:
@silverailscolo Thanks! 

I think switching to the new icon (which is probably more recognizable) through JMRI is the best choice. 

No problem with losing the use of the older icon. We can hang on to that in case it's needed for some time in the future.
